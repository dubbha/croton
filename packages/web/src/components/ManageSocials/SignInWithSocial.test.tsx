import React from 'react';
import { render } from '@testing-library/react';

describe('components/SignInWithSocial', () => {
  const getGenericSocialAPI = () => ({
    onResponseAction,
    onErrorAction,
    errorMessage,
    buttonText,
  }) => (
    <div data-testid="genericSocialApi">
      <div>{onResponseAction}</div>
      <div>{onErrorAction}</div>
      <div>{errorMessage}</div>
      <div>{buttonText}</div>
    </div>
  );

  const setUpMocks = () => {
    jest.doMock('./FbAuth', () => ({
      FbAuth: getGenericSocialAPI(),
    }));

    jest.doMock('./GoogleAuth', () => ({
      GoogleAuth: getGenericSocialAPI(),
    }));

    const { SignInWithSocial } = require('./SignInWithSocial');

    return render(<SignInWithSocial />);
  };

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should render succesfully', () => {
    jest.isolateModules(() => {
      const { container } = setUpMocks();

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
