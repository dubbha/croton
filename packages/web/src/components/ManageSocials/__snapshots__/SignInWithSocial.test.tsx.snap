// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/SignInWithSocial should render succesfully 1`] = `
<div
  class="authButtonContainer"
>
  <div
    data-testid="genericSocialApi"
  >
    <div>
      AUTH_FACEBOOK
    </div>
    <div>
      AUTH_FACEBOOK_ERROR
    </div>
    <div>
      Sorry, something went wrong with logging you in via Facebook
    </div>
    <div />
  </div>
</div>
`;
